### PR TITLE
daemon,o/devicestate: have DeviceManager.SystemMode take an expectation on the system

### PR DESCRIPTION
**Pull request by pedronis**
the expectations control whether to report "run" or "" on non-UC20
systems.

this is an idea discussed with @bboozzoo, it avoids having to resort to
use the cached value field directly sometimes which is confusing.

### CODE REVIEW

File: overlord/devicestate/devicemgr.go [hunk 1]
```diff
@@ -378,21 +380,35 @@ func setClassicFallbackModel(st *state.State, device *auth.DeviceState) error {
...
+	SysAny SysExpectation = iota
+	// SysHasModeenv indicates only systems with modeenv are appropriate.
+	SysHasModeenv
+)
+
+// SystemMode returns the current mode of the system.
+// An expecation about the system should be passed in, if it's SysAny
+// and the system is pre-UC20 where there is no explicitly mode
+// "run" is returned, if it's SysHasModeenv on such systems it returns
+// simply "".
```
Comment (bboozzoo) [inline overlord/devicestate/devicemgr.go:396]:
```suggestion
// An expectation about the system overrides the returned mode
// when none was set explicitly. In which case, with SysAny, the
// mode defaults to implicit "run", thus covering pre-UC20 systems.
// With SysHasModeeenv, there is no implicit default in systems 
// that use modeenv and thus "" is returned.
```

**Review by bboozzoo — APPROVED** [review]
LGTM, Thanks!

**Review by stolowski — APPROVED** [review]
LGTM

**Review by anonymouse64 — APPROVED** [review]
lgtm, but I agree with Maciej's suggestion on the doc-comment

File: overlord/devicestate/devicemgr.go (hunk 1 above)
Comment (pedronis) [reply overlord/devicestate/devicemgr.go]: I applied a tweaked version of this that is even more painstaking.

**Pull request merged**